ItsJonQ/g2
# Incrementally introduce TypeScript language

### Testing instructions

Checkout the branch and run `tsc --build` and ensure there are no errors. Manually check `utils/types` and ensure that they're being built correctly and as expected (especially anything re-exporting lodash).

For some reason this breaks the pre-commit hook in this manner:

```
📦  Zero test...

Configuration error:

Could not locate module @wp-g2/substate/types mapped as:
/Users/sara/projects/g2/packages/substate/types/src.

Please check your configuration for these entries:
{
  "moduleNameMapper": {
    "/^@wp-g2\/(.*)$/": "/Users/sara/projects/g2/packages/$1/src"
  },
  "resolver": null
}
    at createNoMappedModuleFoundError (/Users/sara/projects/g2/node_modules/jest-resolve/build/index.js:501:17)
    at Resolver.resolveStubModuleName (/Users/sara/projects/g2/node_modules/jest-resolve/build/index.js:462:21)
    at Resolver.getMockModule (/Users/sara/projects/g2/node_modules/jest-resolve/build/index.js:304:31)
    at /Users/sara/projects/g2/node_modules/jest-resolve-dependencies/build/index.js:67:45
    at Array.reduce (<anonymous>)
    at DependencyResolver.resolve (/Users/sara/projects/g2/node_modules/jest-resolve-dependencies/build/index.js:53:25)
    at DependencyResolver.resolveInverseModuleMap (/Users/sara/projects/g2/node_modules/jest-resolve-dependencies/build/index.js:175:30)
    at DependencyResolver.resolveInverse (/Users/sara/projects/g2/node_modules/jest-resolve-dependencies/build/index.js:198:17)
    at SearchSource.findRelatedTests (/Users/sara/projects/g2/node_modules/@jest/core/build/SearchSource.js:280:30)
    at SearchSource.findRelatedTestsFromPattern (/Users/sara/projects/g2/node_modules/@jest/core/build/SearchSource.js:345:19)
husky > pre-commit hook failed (add --no-verify to bypass)
```
